lib/spack/docs/packaging_guide_creation.rst

```diff
@@ -777,28 +782,39 @@ The logic behind this sort order is two-fold:
 #. The most-recent development version of a package will usually be newer than any released numeric versions.
    This allows the ``@develop`` version to satisfy dependencies like ``depends_on(abc, when="@x.y.z:")``
 
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-Specifying versions in other directives
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Many Spack directives accept versions speciers, for example
+.. _version_constraints:
```

Comment:
Actually, this section is about defining versions and corresponding source code.

The `depends_on` / `conflicts` directives are yet to be covered.

Maybe this section could be standalone as an introduction to the directives, between `Source code and versions` and `variants`?